setupTest.js

```diff
@@ -1 +1,23 @@
-jest.setTimeout(240000);
+//eslint-disable-next-line node/no-extraneous-require
+const { beforeEach } = require('@jest/globals');
+//eslint-disable-next-line node/no-unpublished-require
+const rimraf = require('rimraf');
+const { join } = require('path');
+
+const directories = ['entry/defaults-index', 'entry/flag-entry', 'entry/multiple-entries', 'hot', 'mode/mode-with-config', 'no-hot'];
+
+jest.setTimeout(300000);
+
+beforeEach(async (done) => {
+    for (const directory of directories) {
+        await rimraf(join(__dirname, `./test/${directory}/bin/*`), () => {
+            done();
+        });
+        await rimraf(join(__dirname, `./test/${directory}/binary/*`), () => {
+            done();
+        });
+        await rimraf(join(__dirname, `./test/${directory}/dist/*`), () => {
+            done();
+        });
+    }
+});
```

Comment:
This will run before each and every test case. Multiple test suites run simultaneously in jest. This can lead to the removal of files when they shouldn't be. 

I think we can overcome this if we can access the directory of the particular test suite when it's running if that is possible.

Reply:
/cc @SimenB, what do you think is this a good idea to inject `beforeEach` globally like this?

Reply:
Use `Promise.all` and avoid using `node`

Reply:
`beforeEach` should be available globally already, no need to import? I might be misunderstanding.

As for the deletion, you should use `del` or something which returns a promise, then `map` over the directories and feed that array into `Promise.all`

(node 14 has https://nodejs.org/api/fs.html#fs_fspromises_rm_path_options, but I assume this needs to work on more node versions than that)